dtfs-central-api/src/v1/controllers/tfm/deal-cancellation/tfm-put-update-deal-cancellation.controller.ts

```diff
@@ -0,0 +1,43 @@
+import { Response } from 'express';
+import { flatten } from 'mongo-dot-notation';
+import { HttpStatusCode } from 'axios';
+import { ApiError, ApiErrorResponseBody, AUDIT_USER_TYPES, CustomExpressRequest } from '@ukef/dtfs2-common';
+import { generateAuditDatabaseRecordFromAuditDetails, validateAuditDetailsAndUserType } from '@ukef/dtfs2-common/change-stream';
+import { UpdateResult } from 'mongodb';
+import { TfmDealCancellationRepo } from '../../../../repositories/tfm-deals-repo';
+import { PutDealCancellationPayload } from '../../../routes/middleware/payload-validation';
+
+type UpdateTfmDealCancellationRequest = CustomExpressRequest<{
+  reqBody: PutDealCancellationPayload;
+}>;
+
+type UpdateTfmDealCancellationResponse = Response<ApiErrorResponseBody | UpdateResult>;
```

Comment:
It's fairly generic, however it's only `PUT` requests that return an `UpdateResult` and this is the first time there's a type that returns `Response<ApiErrorResponseBody | UpdateResult>` so this would be the only use. I'd prefer to keep it as is, and if someone else find they also use the same return type they can pull it out at that point to follow YAGNI principles.